apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -60,7 +60,10 @@ export class TaskUC {
 		// TODO get permitted tasks...
 		// TODO have BL from repo here
 
-		const [tasks, total] = await this.taskRepo.findAllOpenByStudent(userId, pagination);
+		const [submissionsOfStudent, submissionCount] = await this.submissionRepo.getAllSubmissionsByUser(userId);
+		const homeworksWithSubmissions = submissionsOfStudent.map((submission) => submission.task.id);
+
+		const [tasks, total] = await this.taskRepo.findAllByStudent(userId, pagination, homeworksWithSubmissions);
 		const computedTasks = tasks.map((task) => TaskMapper.mapToResponse(task));
```

Comment:
when a service would call this and other usecases, it might want to use task-entities, not the mapped responses of tasks. this mapping must happen in the controller and should not be mixed up with a mapping here (which might be added, but then additionally)

Reply:
I don't agree. Task is a save scope that should not expose any information over the internals. No instance from outside should know information's about the entity's. I my mind this is one of the important points in DDD. But we should speak about it, to get a consent. 
In other case we also must kill all the ...Info Entitys because files, course, coursegroups, user and lessons are other scopes and should not in control of task. The task scope should have no need that this instances exist and it is danger that they know something about the internal data structure.